### ISSUE TITLE
Current UsersController should be renamed and new UsersController introduced

### ISSUE THREAD

**Issue by Kevin-MM** [Submitty/Submitty#5565]
**What problem are you trying to solve with Submitty**
- I think `UsersController` at admin/UsersController.php should be renamed to `CourseController` or perhaps `CourseUsersController`.  A large majority of the functionality inside this controller deals with CRUD operations for users at the course level, not really at the user level.

- As more and more settings are now being configured on a per-user basis (for example time zone and preferred images) I think it would be appropriate to put those routes into a new `UsersController` and remove them from the `HomePageController` where most of those routes are ending up.

**Comment by MasterOdin** [comment]
I agree that we should be splitting apart the controllers more, creating more sub-folders, etc, and making the controllers as focused as possible (which also makes it easier to test a particular controller fully). Can also split apart the ForumController into a ThreadController and PostController for example.
